### ISSUE TITLE
The calendar is missing accessibility properties

### ISSUE THREAD

**Issue by barettjfed** [Hacker0x01/react-datepicker#4518]
**Describe the bug**
According to the [ARIA Authoring Practices guide for W3C](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/) the calendar popup should have `role="dialog"` and `aria-modal="true"`. Without these in place screen readers will not have trapped focus.

**To Reproduce**

1. Go to any react-datepicker example
2. Notice that they are missing these properties

**Expected behavior**
role="dialog" and aria-modal="true" are present on the calendar popup.
